packages/devtools_app/lib/src/performance/timeline_flame_chart.dart

```diff
@@ -42,89 +45,94 @@ class TimelineFlameChartContainer extends StatefulWidget {
 
   @override
   Widget build(BuildContext context) {
     Widget content;
-    final timelineEmpty = (controller.data?.isEmpty ?? true) ||
-        controller.data.eventGroups.isEmpty;
-    if (widget.processing) {
-      content = _buildProcessingInfo();
-    } else if (timelineEmpty) {
-      content = Center(
-        key: TimelineFlameChartContainer.emptyTimelineKey,
-        child: Row(
-          mainAxisAlignment: MainAxisAlignment.center,
-          children: [
-            Text(
-              'No timeline events. Try clicking the refresh button ',
-              style: Theme.of(context).subtleTextStyle,
-            ),
-            Icon(
-              Icons.refresh,
-              size: defaultIconSize,
-            ),
-            Text(
-              ' to load more data.',
-              style: Theme.of(context).subtleTextStyle,
-            ),
-          ],
-        ),
-      );
+    if (frameAnalysisSupported && selectedTab != null) {
+      content = FlutterFrameAnalysisView(frame: selectedTab.frame);
     } else {
-      content = LayoutBuilder(
-        builder: (context, constraints) {
-          return TimelineFlameChart(
-            controller.data,
-            width: constraints.maxWidth,
-            height: constraints.maxHeight,
-            selectionNotifier: controller.selectedTimelineEvent,
-            searchMatchesNotifier: controller.searchMatches,
-            activeSearchMatchNotifier: controller.activeSearchMatch,
-            onDataSelected: (e) => controller.selectTimelineEvent(e),
-          );
-        },
-      );
+      final timelineEmpty = (controller.data?.isEmpty ?? true) ||
+          controller.data.eventGroups.isEmpty;
+      if (widget.processing) {
+        content = _buildProcessingInfo();
+      } else if (timelineEmpty) {
+        content = Center(
+          key: TimelineAnalysisContainer.emptyTimelineKey,
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              Text(
+                'No timeline events. Try clicking the refresh button ',
+                style: Theme.of(context).subtleTextStyle,
+              ),
+              Icon(
+                Icons.refresh,
+                size: defaultIconSize,
+              ),
+              Text(
+                ' to load more data.',
+                style: Theme.of(context).subtleTextStyle,
+              ),
+            ],
+          ),
+        );
+      } else {
+        content = LayoutBuilder(
+          builder: (context, constraints) {
+            return TimelineFlameChart(
+              controller.data,
+              width: constraints.maxWidth,
+              height: constraints.maxHeight,
+              selectionNotifier: controller.selectedTimelineEvent,
+              searchMatchesNotifier: controller.searchMatches,
+              activeSearchMatchNotifier: controller.activeSearchMatch,
+              onDataSelected: (e) => controller.selectTimelineEvent(e),
```

Comment:
Can this just be `onDataSelected: controller.selectTimelineEvent` ?

Reply:
yep - done